Scala/Scala.sublime-syntax

```diff
@@ -440,16 +578,123 @@ contexts:
+    - match: '{{typeprefix}}'
+      push:
+        - match: '(?=[\B\s]=[\B\s])'
+          pop: true
+        - include: delimited-type-expression
+        - match: '(?=[\{\n])'
+          pop: true
+    - include: val-pattern-match-main
+    - match: \(
+      push:
+        - match: \)
+          pop: true
+        - include: val-pattern-match-inner
+  val-pattern-match-inner:
+    - include: val-pattern-match-main
     - match: '{{upperid}}'
       scope: support.class.scala
+
+  pattern-match:
+    - include: keywords
+    - include: constants
+    - include: char-literal
+    - include: scala-symbol
+    - include: strings
+    - include: xml-literal
+    - match: '`'
+      push:
+        - match: '`'
+          pop: true
+    - match: '{{varid}}\.'
+    - match: '\.{{varid}}'
+    - match: '\b{{varid}}'
+      scope: variable.parameter.scala   # not indexed!
+    - include: ascription
     - match: \[
       push:
         - match: \]
           pop: true
-        - include: main
+        - include: delimited-type-expression
     - match: '@|_'
       scope: keyword.other.scala
+
+  base-type-expression:
+    - match: '\(\{\s*type\s+λ\[α(\[_\])?(,\s*β(\[_\])?)?\]\s*='
+      scope: comment.block.scala
+      push:
+        - match: '\}\)#λ'
+          scope: comment.block.scala
+          pop: true
+        - include: delimited-type-expression
+    - match: \(
+      push:
+        - match: \)
+          pop: true
+        - include: delimited-type-expression
+    - match: \[
+      push:
+        - match: \]
+          pop: true
+        - include: delimited-type-expression
+    - match: \{
+      push:
+        - match: \}
+          pop: true
+        - include: main
+  type-constraints:
+    - match: '{{oplookbehind}}(<:|>:|<%|\+|-|:){{oplookahead}}'
+      scope: keyword.operator
+  delimited-type-expression:
+    - match: "[αβ]"     # just here for type lambdas
+      scope: comment.block.empty.scala
+    - match: '[\.#]'
+      scope: punctuation.separator
+    - include: base-types
+    - match: '\b(forSome)\b'
+      scope: keyword.declaration.scala
+    - match: '{{upperid}}'
+      scope: support.class.scala
+    - match: _
+    - match: '{{id}}'
+      scope: support.type.scala
+    - include: base-type-expression
+  single-type-expression:
+    - match: "[αβ]"     # just here for type lambdas
+      scope: comment.block.empty.scala
+    - match: \b(Unit|Boolean|Byte|Char|Short|Int|Float|Long|Double)\b
+      scope: storage.type.primitive.scala
+      set: single-type-expression-tail
+    - match: '\b(forSome)\b'
+      scope: keyword.declaration.scala
+    - match: '{{upperid}}'
+      scope: support.class.scala
+      set: single-type-expression-tail
+    - match: '{{id}}'
+      scope: support.type.scala
+      set: single-type-expression-tail
+    - include: base-type-expression
+    - match: '(?=[\s,\)\}\]])'
+      pop: true
+  single-type-expression-tail:
+    - match: '[\.#]'
+      scope: punctuation.separator
```

Comment:
This should probably be `punctuation.accessor.scala`, or at least needs `.scala` at the end